### PR TITLE
Fixes incomplete snippets with expand query options and filter query options that use function operator types

### LINKED ISSUE #1626: [CLI] Wrong snippets generated for http snippets with filter query option that use Functions operator type.
**Bug**
The CLI snippet generator is giving out wrong snippets for http snippets that uses Functions as the operator type for filter query options.  Such snippets eventually fail to execute on raptor with the below error.
More information on operators and functions can be found [here](https://learn.microsoft.com/en-us/graph/filter-query-parameter?tabs=http)
image

**Actual generated snippet (Example)**
``mgc identity-governance access-reviews definitions list --filter contains(scope/microsoft.graph.accessReviewQueryScope/query,'./members')``

**Expected snippet (Example)**
``mgc identity-governance access-reviews definitions list --filter "contains(scope/microsoft.graph.accessReviewQueryScope/query, './members')"``

### LINKED ISSUE #1627: [CLI]Incomplete snippet generation for http snippets with expand query options
**Bug**
Inexecutable cli snippet are generated for the below http path.
GET /me/messages/{id}?$expand=singleValueExtendedProperties($filter=id eq '{id_value}')
**Actual snippet**
``mgc me messages get --message-id {message-id} --expand singleValueExtendedProperties($filter``
**Expected snippet**
``mgc me messages get --message-id {message-id} "singleValueExtendedProperties(`$filter=id eq '{id_value}')"``

### PR DESCRIPTION
**Pull request by timayabi2020**
## Overview

Fixes #1627 
Fixes #1626

## Testing Instructions

* Checkout branch associated with this PR
* Run it locally and formulate the request as per the screenshots below showing the request and expected output.
image

image

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs [hunk 1]
```diff
@@ -220,16 +221,29 @@ private static void ProcessMeSegments(List<string> commandSegments, string opera
...
+                string queryOptionFunction = splittedQueryString[1].Replace("$", "`$");
+                queryOptionFunction = queryOptionFunction.Replace(queryOptionFunction, "\"" + queryOptionFunction + "\"");
+                splitQueryString.Add(queryOption, queryOptionFunction);
+            }
+            else
+            {
+                splitQueryString = snippetModel.QueryString
+                        .Remove(0, 1)
+                        .Split('&')
+                        .Select(q =>
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs:239]:
```suggestion
                        .Select(static q =>
```

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs [hunk 2]
```diff
@@ -220,16 +221,29 @@ private static void ProcessMeSegments(List<string> commandSegments, string opera
...
+                        {
+                            var x = q.Split('=');
+                            return x.Length > 1 ? (x[0], x[1]) : (x[0], string.Empty);
+                        })
+                        .Where(t => !string.IsNullOrWhiteSpace(t.Item2))
+                        .ToDictionary(t => t.Item1, t => t.Item2);
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs:245]:
```suggestion
                        .Where(static t => !string.IsNullOrWhiteSpace(t.Item2))
                        .ToDictionary(static t => t.Item1, static t => t.Item2);
```

**Review by baywet — CHANGES_REQUESTED** [review]
FYI the syntax for GitHub is `fixes #13099` (one per line, no comma etc...)

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs [hunk 3]
```diff
@@ -220,16 +221,29 @@ private static void ProcessMeSegments(List<string> commandSegments, string opera
...
-                        return x.Length > 1 ? (x[0], x[1]) : (x[0], string.Empty);
-                    })
-                    .Where(t => !string.IsNullOrWhiteSpace(t.Item2))
-                    .ToDictionary(t => t.Item1, t => t.Item2);
+            if (systemQueryOptionRegex.IsMatch(snippetModel.QueryString))
+            {
+                string pattern = "\\?\\$\\w*=";
+                string[] splittedQueryString = Regex.Split(snippetModel.QueryString, pattern, RegexOptions.Compiled, TimeSpan.FromSeconds(5));
+                var match = Regex.Match(snippetModel.QueryString, pattern, RegexOptions.Compiled, TimeSpan.FromSeconds(5));
+                string queryOption = match.Groups[0].Value.Replace("?", "").Replace("=", "");
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs:229]:
```suggestion
                string queryOption = match.Groups[0].Value.Replace("?", string.Empty, StringComparison.OrdinalIgnoreCase).Replace("=", string.Empty, StringComparison.OrdinalIgnoreCase);
```

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs [hunk 4]
```diff
@@ -220,16 +221,29 @@ private static void ProcessMeSegments(List<string> commandSegments, string opera
...
+                string queryOptionFunction = splittedQueryString[1].Replace("$", "`$");
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs:230]:
```suggestion
                string queryOptionFunction = splittedQueryString[1].Replace("$", "`$", StringComparison.OrdinalIgnoreCase);
```

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs [hunk 5]
```diff
@@ -220,16 +221,29 @@ private static void ProcessMeSegments(List<string> commandSegments, string opera
...
+                queryOptionFunction = queryOptionFunction.Replace(queryOptionFunction, "\"" + queryOptionFunction + "\"");
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs:231]:
```suggestion
                queryOptionFunction = queryOptionFunction.Replace(queryOptionFunction, "\"" + queryOptionFunction + "\"", StringComparison.Ordinal);
```

**Review by baywet — CHANGES_REQUESTED** [review]

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs (hunk 3 above)
Comment (timayabi2020) [reply CodeSnippetsReflection.OpenAPI/LanguageGenerators/GraphCliGenerator.cs:229]: @baywet  thanks for these corrections

**Review by baywet — APPROVED** [review]

**Pull request merged**